### ISSUE TITLE
Init of Trivy from the `main` branch is slow.

### ISSUE THREAD

**Issue by DmitriyLewen** [aquasecurity/trivy#7431]
## Description
After #7417 Trivy starts slower.
```bash
➜ git checkout a5aa63eff7e229744090f9ad300c1bec3259397e
Previous HEAD position was 39c80248b chore(deps): Bump trivy-checks (#7417)
HEAD is now at a5aa63eff fix(misconf): do not register Rego libs in checks registry (#7420)
➜ mage build
➜ time ./trivy -v 
./trivy -v  0.35s user 0.05s system 30% cpu 1.304 total

➜ git checkout 39c80248bcafd296e0bb6712113da3abe70e4ce2
Previous HEAD position was a5aa63eff fix(misconf): do not register Rego libs in checks registry (#7420)
HEAD is now at 39c80248b chore(deps): Bump trivy-checks (#7417)
➜ mage build 
➜ time ./trivy -v                                      
./trivy -v  8.58s user 0.19s system 129% cpu 6.791 total
```

**Comment by DmitriyLewen** [comment]
This is critical for us.
We are delaying the release of `v0.55.0` until this issue is resolved.

**Comment by simar7** [comment]
This is due two reasons:

1. (short term fix): There is a performance issue in OPA. See [here](https://github.com/open-policy-agent/opa/issues/6969) for the issue. We have a [fix](https://github.com/open-policy-agent/opa/pull/6970) for it upstream but it hasn't been merged just yet. In the meantime, we can cherry pick the fix to  not delay our release as done [here](https://github.com/aquasecurity/trivy/pull/7427/commits/b2b3f92d414e188ec419bc5ec0b7a09540a2e62b#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R424)

2. (long term fix, separate issue) As @knqyf263 mentioned [here](https://github.com/aquasecurity/trivy/issues/7426#issuecomment-2323929191) we should only initialize the misconf checks when the misconfiguration scanner is enabled. We will tackle this improvement in a separate feature improvement so we don't delay the current release.
